subworkflows/local/align_pacbio.nf

```diff
@@ -71,15 +83,49 @@ workflow ALIGN_PACBIO {
     CREATE_CRAM_FILTER_INPUT ( GENERATE_CRAM_CSV.out.csv, fasta )
     ch_versions = ch_versions.mix ( CREATE_CRAM_FILTER_INPUT.out.versions )
 
-    // Filter BAM and output as FASTQ
-    FILTER_PACBIO ( CREATE_CRAM_FILTER_INPUT.out.chunked_cram, db )
-    ch_versions = ch_versions.mix ( FILTER_PACBIO.out.versions )
+    //
+    // FILTER BAMs AND OUTPUT AS FASTQ
+    //
+    CREATE_CRAM_FILTER_INPUT.out.chunked_cram
+    | map { meta, cram -> [ meta, cram, [] ] }
+    | set { ch_pacbio }
+
+    SAMTOOLS_CONVERT ( ch_pacbio, [ [], [] ], [] )
+    ch_versions = ch_versions.mix ( SAMTOOLS_CONVERT.out.versions )
+
+    if ( params.filter_pacbio ) {
+        // Collate BAM file to create interleaved FASTA
+        SAMTOOLS_COLLATETOFASTA ( SAMTOOLS_CONVERT.out.bam )
+
+        BLASTN_HIFI ( SAMTOOLS_COLLATETOFASTA.out.fasta, hifi_adapter_db )
+        BGZIP_BLASTN ( BLASTN_HIFI.out.txt )
 
-    FASTQC_FILTERED ( FILTER_PACBIO.out.fastq )
+        bam_blast = SAMTOOLS_CONVERT.out.bam.join ( BGZIP_BLASTN.out.output )
+
+        // TO FIX: hifi_trimmer only run one when using channel hifi_adapter_yaml provided as input
+        HIFI_TRIMMER ( bam_blast, params.hifi_adapter_yaml )
+
+        ch_reads_for_align =  HIFI_TRIMMER.out.fastq
+
+        // FastQC on filtered reads
+        FASTQC_FILTERED ( ch_reads_for_align )
+        ch_post_qc = ch_post_qc.mix ( FASTQC_FILTERED.out.zip )
+
+        ch_versions = ch_versions
+        | mix ( SAMTOOLS_COLLATETOFASTA.out.versions )
+        | mix ( BLASTN_HIFI.out.versions )
+        | mix ( BGZIP_BLASTN.out.versions )
+        | mix ( HIFI_TRIMMER.out.versions )
+        | mix ( FASTQC_FILTERED.out.versions )
```

Comment:
AFAIK this can cause linting problems - the nf-core linter expects individual `.mix()` calls for each process.